lib/lte_link_control/lte_lc.c

```diff
@@ -346,6 +686,35 @@ int lte_lc_init_and_connect(void)
 	return err;
 }
 
+int lte_lc_connect_async(lte_lc_evt_handler_t handler)
+{
+	if (handler) {
+		evt_handler = handler;
+	} else if (evt_handler == NULL) {
+		LOG_WRN("No handler registered, events will not be received");
```

Comment:
Is this a real use-case. How do we know if we are connected if we used the non-blocking API without an event handler?

Reply:
Probably not a real use-case. Changed to error out here if no handler is registered.